### PR TITLE
eth transaction

### PR DESCRIPTION
**Pull request by nvxtien**

### CODE REVIEW

File: src/main/java/io/tatum/wallet/Address.java [hunk 1]
```diff
@@ -76,27 +63,33 @@ public String generateBchAddress(Boolean testnet, String xpub, int i) throws Exe
...
-
-            DeterministicKey parent = AddressBuilder.build().network(network).lazyFromBase58(xpub);
-
-            DeterministicKey key = AddressBuilder.build().network(network).fromBase58(xpub).lazyDerivePath(path);
-
-            String serializePubB58 = key.serializePubB58(network);
-
-            System.out.println(serializePubB58.toLowerCase());
+            ChildNumber path = new ChildNumber(i, false);
+            NetworkParameters network = BITCOIN_MAINNET;
```
Comment (tatumio) [inline src/main/java/io/tatum/wallet/Address.java:75]: This should be really BITCOIN_MAINNET?

File: src/main/java/io/tatum/wallet/Address.java [hunk 2]
```diff
@@ -76,27 +63,33 @@ public String generateBchAddress(Boolean testnet, String xpub, int i) throws Exe
...
+     *
+     * @param testnet testnet or mainnet version of address
+     * @param xpub    extended public key to generate address from
+     * @param i       derivation index of address to generate. Up to 2^32 addresses can be generated.
+     * @returns blockchain address
+     */
+    public String generateVetAddress(Boolean testnet, String xpub, int i) throws ExecutionException, InterruptedException {
+        return CompletableFuture.supplyAsync(() -> {
+            ChildNumber path = new ChildNumber(i, false);
+            NetworkParameters network = BITCOIN_MAINNET;
```
Comment (tatumio) [inline src/main/java/io/tatum/wallet/Address.java:91]: Same here...

**Review by tatumio — CHANGES_REQUESTED** [review]

**Comment by nvxtien** [comment]
ethereum transaction

File: src/main/java/io/tatum/wallet/Address.java (hunk 1 above)
Comment (nvxtien) [reply src/main/java/io/tatum/wallet/Address.java:75]: Actually, we just xpub prefix for the ethereum extended key. So we can use the network parameters of bitcoin. We define a specific network parameter for ethereum later.

File: src/main/java/io/tatum/wallet/Address.java (hunk 2 above)
Comment (nvxtien) [reply src/main/java/io/tatum/wallet/Address.java:91]: The same as the above comment.

**Pull request merged**